roles/openshift_facts/library/openshift_facts.py

```diff
@@ -901,10 +901,24 @@ def set_sdn_facts_if_unset(facts, system_facts):
             facts['common']['sdn_network_plugin_name'] = plugin
 
     if 'master' in facts:
-        if 'sdn_cluster_network_cidr' not in facts['master']:
-            facts['master']['sdn_cluster_network_cidr'] = '10.1.0.0/16'
-        if 'sdn_host_subnet_length' not in facts['master']:
-            facts['master']['sdn_host_subnet_length'] = '8'
+        # set defaults for sdn_cluster_network_cidr and sdn_host_subnet_length
+        # these might be overridden if they exist in the master config file
+        facts['master']['sdn_cluster_network_cidr'] = '10.128.0.0/14'
+        facts['master']['sdn_host_subnet_length'] = '9'
+
+        master_cfg_path = os.path.join(facts['common']['config_base'],
+                                       'master/master-config.yaml')
+        if os.path.isfile(master_cfg_path):
+            with open(master_cfg_path, 'r') as master_cfg_f:
+                config = yaml.safe_load(master_cfg_f.read())
+
+            if 'networkConfig' in config:
+                if 'clusterNetworkCIDR' in config['networkConfig']:
+                    facts['master']['sdn_cluster_network_cidr'] = \
+                        config['networkConfig']['clusterNetworkCIDR']
+                if 'hostSubnetLength' in config['networkConfig']:
+                    facts['master']['sdn_host_subnet_length'] = \
+                        config['networkConfig']['hostSubnetLength']
```

Comment:
With this implementation there's a chance that the config file has different values from what's in the inventory. We could try to throw an exception but I think it's alright to just accept what's set in inventory variables.